statsd/test_helpers_test.go

```diff
@@ -0,0 +1,411 @@
+package statsd
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testTelemetryData struct {
+	gauge         int
+	count         int
+	histogram     int
+	distribution  int
+	set           int
+	timing        int
+	event         int
+	service_check int
+
+	aggregated_context      int
+	aggregated_gauge        int
+	aggregated_set          int
+	aggregated_count        int
+	aggregated_histogram    int
+	aggregated_distribution int
+	aggregated_timing       int
+
+	metric_dropped_on_receive int
+	packets_sent              int
+	packets_dropped           int
+	packets_dropped_queue     int
+	packets_dropped_writer    int
+	bytes_sent                int
+	bytes_dropped             int
+	bytes_dropped_queue       int
+	bytes_dropped_writer      int
+}
+
+// testServer acts as a fake server and keep track of what was sent to a client. This allows end-to-end testing of the
+// dogstatsd client
+type testServer struct {
+	sync.Mutex
+
+	conn      io.ReadCloser
+	data      []string
+	errors    []string
+	proto     string
+	addr      string
+	stopped   chan struct{}
+	tags      string
+	namespace string
+
+	devMode             bool
+	aggregation         bool
+	extendedAggregation bool
+	telemetry           testTelemetryData
+	telemetryEnabled    bool
+}
+
+const (
+	devModeEnabled = true
+	devModeDisable = false
+
+	aggregationEnabled  = true
+	aggregationDisabled = false
+
+	extendedAggregationEnabled  = true
+	extendedAggregationDisabled = false
+
+	telemetryEnabled  = true
+	telemetryDisabled = false
+)
```

Comment:
what are these constants for? (they seem to be unused)

Reply:
Nowhere anymore :+1: